### PR TITLE
add getActorLikes

### PR DESCRIPTION
**Pull request by estrattonbailey**
We use the term actor in the errors/params, and since a Like could originate from a bot or human, decided `actor` the better term.

### CODE REVIEW

**Review by dholms — COMMENTED** [review]
yup schema looks right 👍

File: packages/bsky/src/api/app/bsky/feed/getActorLikes.ts [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+
+      if (viewer !== null) {
+        feedItemsQb = feedItemsQb
+          .where((qb) =>
+            qb.where((qb) =>
+              graphService.whereNotMuted(qb, viewer, [ref('post.creator')]),
+            ),
+          )
+          // TODO do we want this? was missing here
+          .whereNotExists(graphService.blockQb(viewer, [ref('post.creator')]))
```
Comment (estrattonbailey) [inline packages/bsky/src/api/app/bsky/feed/getActorLikes.ts:64]: This clause was present in the PDS but not here. Was there a reason we didn't want to filter out posts from blocked users?

File: packages/bsky/src/api/app/bsky/feed/getActorLikes.ts [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+        keyset,
+      })
+
+      const [feedItems, repoRev] = await Promise.all([
+        feedItemsQb.execute(),
+        actorService.getRepoRev(viewer),
+      ])
+      setRepoRev(res, repoRev)
+
+      const feed = await feedService.hydrateFeed(feedItems, viewer)
```
Comment (estrattonbailey) [inline packages/bsky/src/api/app/bsky/feed/getActorLikes.ts:84]: This assembles a feed that includes replies/threads. I'm assuming we don't want that for this view. Do we have an existing hydrator for this, or should I just add some options to `hydrateFeed`?

(Same question in PDS)

File: packages/bsky/src/api/app/bsky/feed/getActorLikes.ts [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+      }
+
+      let feedItemsQb = feedService
+        .selectFeedItemQb()
+        .innerJoin('like', 'like.subject', 'feed_item.uri')
+        .where('like.creator', '=', actorDid)
+
+      if (viewer !== null) {
+        feedItemsQb = feedItemsQb
+          .where((qb) =>
```
Comment (dholms) [inline packages/bsky/src/api/app/bsky/feed/getActorLikes.ts:39]: don't think this nested where clause is necessary - you probably had an OR here at some point or something

File: packages/bsky/src/api/app/bsky/feed/getActorLikes.ts (hunk 2 above)
Comment (dholms) [reply packages/bsky/src/api/app/bsky/feed/getActorLikes.ts:64]: I think this is fine 👍 

It's the same hydration that occurs on `getAuthorFeed`

The "reply" property is a bit misleading, it should really be "replyTo" and just provides some context about the parent post

File: packages/bsky/src/api/app/bsky/feed/getActorLikes.ts (hunk 3 above)
Comment (dholms) [reply packages/bsky/src/api/app/bsky/feed/getActorLikes.ts:39]: (same thing in PDS)

File: packages/pds/src/app-view/api/app/bsky/feed/getActorLikes.ts [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+import { paginate } from '../../../../../db/pagination'
+import AppContext from '../../../../../context'
+import { FeedRow } from '../../../../services/feed'
+import { OutputSchema } from '../../../../../lexicon/types/app/bsky/feed/getAuthorFeed'
+import { handleReadAfterWrite } from '../util/read-after-write'
+import { authPassthru } from '../../../../../api/com/atproto/admin/util'
+import { LocalRecords } from '../../../../../services/local'
+
+export default function (server: Server, ctx: AppContext) {
+  server.app.bsky.feed.getActorLikes({
```
Comment (dholms) [inline packages/pds/src/app-view/api/app/bsky/feed/getActorLikes.ts:13]: good news: we're incredibly close to not having to re-implement these on the PDS 👀

**Review by dholms — COMMENTED** [review]
couple small things but overall looking good!

File: packages/pds/src/app-view/api/app/bsky/feed/getActorLikes.ts [hunk 5]
```diff
@@ -0,0 +1,135 @@
...
+              localProf.record,
+            ),
+          },
+        }
+      } else {
+        return item
+      }
+    })
+  }
+  feed = await localSrvc.formatAndInsertPostsInFeed(feed, local.posts)
```
[the remark this answers is not in the document]
Comment (estrattonbailey) [reply packages/pds/src/app-view/api/app/bsky/feed/getActorLikes.ts]: Cool yeah wasn't sure on this

**Comment by gildaswise** [comment]
Hey, what's the reason for `9b1f5f0` which disallows seeing other profile's likes? Shouldn't that be something each client decides? The statement was "Those likes can still be accessed by APIs, so other apps may show them."

**Comment by estrattonbailey** [comment]
Good question, and totally understand the confusion. The distinction here is that Like _records_ are public i.e. they are publicized via our BGS, but we chose not to provide access to that data from our App View.

Clients can opt to interact with our App View APIs, or choose make their own set of APIs if they want to make different decisions that we have. Some users have already done this but ingesting Like records from our BGS and providing them as custom feeds.

**Pull request merged**